server/client.go

```diff
@@ -2050,6 +2079,40 @@ func (c *client) queueOutbound(data []byte) {
 	// Add to pending bytes total.
 	c.out.pb += int64(len(data))
 
+	// Take a copy of the slice ref so that we can chop bits off the beginning
+	// without affecting the original "data" slice.
+	toBuffer := data
+
+	// All of the queued []byte have a fixed capacity, so if there's a []byte
+	// at the tail of the buffer list that isn't full yet, we should top that
+	// up first. This helps to ensure we aren't pulling more []bytes from the
+	// pool than we need to.
+	if len(c.out.nb) > 0 {
+		last := &c.out.nb[len(c.out.nb)-1]
```

Comment:
Why does this need to be addr? Seems its re-assigned below but seemingly with no effect?

Reply:
This is the coalescing in action — at this point, `c.out.nb` might still have things queued up from a previous write, so we take the reference to the last buffer, see whether there's any remaining space that we can "top up" with this write and append that many bytes onto the existing buffer before we allocate a new one.

i.e. if we have an 8K buffer, the last write was only 7K, we have 1K left, so we can write another 1K onto the end of that buffer before we need to pull another one out of the pool.